models/demos/wormhole/mamba/tt/cache.py

```diff
@@ -52,31 +52,23 @@ def get(self, user_idx: int, entry_idx: int) -> ttnn.Tensor:
             self.cache[entry_idx][user_idx], device=self.device, memory_config=self.cache_memory_config
         )
 
-    def concat_users(self, entry_idx: int, layout=ttnn.TILE_LAYOUT):
+    def concat_users(self, entry_idx: int, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG):
         assert entry_idx < len(self.cache), f"Expected key {entry_idx} to exist in cache"
         values = self.cache[entry_idx]
         if self.on_host:
             values = [
-                ttnn.to_device(values[i], device=self.device, memory_config=self.cache_memory_config)
+                ttnn.to_device(values[i], device=self.device, memory_config=memory_config)
                 for i in range(self.num_users)
             ]
         return ttnn.to_layout(ttnn.concat(values, dim=2), layout)
 
     def reset(self):
         for entry_idx in range(len(self.cache)):
             for user_idx in range(self.num_users):
-                ttnn.deallocate(self.cache[entry_idx][user_idx])
```

Comment:
This deallocation was there to make sure old cache was garbage collected explictly